python/cudf/cudf/core/column/categorical.py

```diff
@@ -1417,8 +1417,17 @@ def _get_decategorized_column(self) -> ColumnBase:
 
     def copy(self, deep: bool = True) -> CategoricalColumn:
         if deep:
+
             copied_col = libcudf.copying.copy_column(self)
             copied_cat = libcudf.copying.copy_column(self.dtype._categories)
+            if is_interval_dtype(self.dtype._categories):
+                # avoiding circular import since IntervalColumn
+                # inherits from StructColumn.
+                from cudf.core.column import IntervalColumn
+
+                copied_cat = IntervalColumn.from_struct_column(
```

Comment:
```suggestion
                copied_cat = column.IntervalColumn.from_struct_column(
```

You could just use `column` which is already imported.